src/view.js

```diff
@@ -0,0 +1,223 @@
+var c = require("./ttConstants.js");
+var GridBoard = require("./board/gridBoard.js");
+
+function View(game, turnMap) {
+  this.game = game;
+  this.turnMap = turnMap;
+  this.turnMap.updateState("start");
+  this.tokenViews = [];
+  this.tileViews = [];
+  this.boardView = new PIXI.Graphics();
+  this.stage = new PIXI.Container();
+  this.renderer = PIXI.autoDetectRenderer(c.canvasWidth, c.canvasHeight, 
+                                          { transparent: true });
+  document.body.appendChild(this.renderer.view);
+  
+  if (game.board instanceof GridBoard) { 
+    for (var i = 0; i < this.game.board.height; i++) { 
+      this.tileViews[i] = [];
+    } 
+  } 
+};
+
+View.prototype.drawBoard = function() { 
+  if (this.game.board instanceof GridBoard) 
+    this.drawGridBoard();
+  
+  /* todo: 
+
+   else if (this.game.board instanceof PathBoard) 
+     this.drawPathBoard();
+   else if (this.game.board instanceof GraphBoard)
+     this.drawGraphBoard();
+   else 
+     do nothing and defer drawing to user? 
+
+   */
+};
+
+View.prototype.drawGridBoard = function() {
+  
+  this.boardView.x = c.boardStartX;
+  this.boardView.y = c.boardStartY;
+  this.boardView.beginFill(c.blueColor, 1);
+  this.boardView.drawRect(0, 0, c.boardWidth, c.boardHeight);
+  this.stage.addChild(this.boardView);
+  this.drawTiles(this.boardView);
+  this.drawTokens();
+  this.drawMessage();
+  this.animate();
+};
+
+// todo 
+View.prototype.drawPathBoard = function() {
+
+};
+
+// todo 
+View.prototype.drawGraphBoard = function() {
+
+};
+
+View.prototype.drawTile = function(opts) { 
+  var tileView = new PIXI.Graphics();
+  tileView.lineStyle(1, 0, 1);
+  tileView.beginFill(opts.tile.color, 1);
+  tileView.drawRect(0, 0, opts.dim.width, opts.dim.height);
+  tileView.x = opts.position.x;
+  tileView.y = opts.position.y;
+  
+  if (this.game.moveType == c.moveTypeManual) { 
+    tileView.interactive = true;
+    var context = this;
+    tileView.click = function(mouseData) {
+      /* make sure we're in the right state, 
+         a token has been pressed, 
+         and we're not a tile with a token on it (if we have > 0
+         children, then this click was meant for a token... */
+      if (context.turnMap.getCurrentState() == "waitingForMove" && 
```

Comment:
I think we should be able to just say that the tile was clicked and send the action to the game. Then the game can decide what do to based on that action depending on state. That also removes the need for `if (this.game.moveType == c.moveTypeManual` because we can always allow clicking on tiles and the game can handle what to do with the click